tests/kernel/timer/timer_error_case/src/main.c

```diff
@@ -0,0 +1,385 @@
+/*
+ * Copyright (c) 2020 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <stdlib.h>
+#include <ztest.h>
+#include <zephyr/types.h>
+#include <irq_offload.h>
+#include <ztest_error_hook.h>
+
+#define STACK_SIZE (512 + CONFIG_TEST_EXTRA_STACKSIZE)
+#define THREAD_TEST_PRIORITY 0
+#define TEST_TIMEOUT -20
+#define PERIOD 50
+#define DURATION 100
+
+static struct k_timer mytimer, sync_timer;
+static struct k_thread tdata;
+static K_THREAD_STACK_DEFINE(tstack, STACK_SIZE);
+
+#ifdef CONFIG_USERSPACE
+static void thread_timer_start_null(void *p1, void *p2, void *p3)
+{
+	ztest_set_fault_valid(true);
+	k_timer_start(NULL, K_MSEC(DURATION), K_NO_WAIT);
+
+	/* should not go here*/
+	ztest_test_fail();
+}
+
+/**
+ * @brief Test k_timer_start() API
+ *
+ * @details Create a thread and set k_timer_start() input to NULL
+ *
+ * @ingroup kernel_timer_tests
+ *
+ * @see k_timer_start()
+ */
+void test_timer_start_null(void)
+{
+	k_tid_t tid = k_thread_create(&tdata, tstack, STACK_SIZE,
+			(k_thread_entry_t)thread_timer_start_null,
+			NULL, NULL, NULL,
+			K_PRIO_PREEMPT(THREAD_TEST_PRIORITY),
+			K_USER | K_INHERIT_PERMS, K_NO_WAIT);
+
+	k_thread_join(tid, K_FOREVER);
+}
+#else
+void test_timer_start_null(void)
+{
+	/* For those platform not support userspace, we skip it. */
+	ztest_test_skip();
+}
+#endif
+
+#ifdef CONFIG_USERSPACE
+static void thread_timer_stop_null(void *p1, void *p2, void *p3)
+{
+	ztest_set_fault_valid(true);
```

Comment:
Please remove all codes of testing assert().

Reply:
@LeiW000 excuse me, this is for testing Z_OOPS, not ASSERT, so I think it's ok to keep it, thanks!

Reply:
Yes, I have removed all testing of assert(). Those cases verify for oops.